src/test/java/tatracker/model/group/GroupTest.java

```diff
@@ -0,0 +1,178 @@
+package tatracker.model.group;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static tatracker.testutil.group.TypicalGroups.MANY_STUDENTS;
+import static tatracker.testutil.group.TypicalGroups.MANY_STUDENTS_COPY;
+import static tatracker.testutil.group.TypicalGroups.NO_STUDENTS;
+import static tatracker.testutil.group.TypicalGroups.ONE_STUDENT;
+
+import org.junit.jupiter.api.Test;
+
+import tatracker.model.student.Matric;
+import tatracker.testutil.student.TypicalStudents;
+
+public class GroupTest {
+    @Test
+    public void equals() {
+        //Same object --> True
+        assertTrue(ONE_STUDENT.equals(ONE_STUDENT));
+
+        //Same group code --> True
+        assertTrue(MANY_STUDENTS.equals(MANY_STUDENTS_COPY));
+
+        //Different groups --> False
+        assertFalse(NO_STUDENTS.equals(MANY_STUDENTS));
+    }
+
+    @Test
+    public void setIdentifier() {
+        Group group = new Group("G03");
+        group.setIdentifier("G04");
+
+        //Identifier value should change
+        assertTrue(group.getIdentifier().equals("G04"));
+    }
+
+    @Test
+    public void setGroupType() {
+        Group group = new Group("G03", GroupType.LAB);
+        group.setGroupType(GroupType.TUTORIAL);
+
+        //GroupType should change
+        assertTrue(group.getGroupType().equals(GroupType.TUTORIAL));
+    }
+
+    @Test
+    public void sortStudentsAlphabetically() {
+        Group group = MANY_STUDENTS;
+        group.sortStudentsAlphabetically();
+
+        //First Student
+        assertEquals(group.get(0), TypicalStudents.ALICE);
+
+        //Middle Student
+        assertEquals(group.get(2), TypicalStudents.DANIEL);
+
+        //Last Student
+        assertEquals(group.get(5), TypicalStudents.GEORGE);
+    }
```

Comment:
Hmmm actually you would need to ensure the entire list matches.
Also since group is not immutable you are modifying the MANY_STUDENTS group constant
Maybe create a deep copy of the constant instead.

Reply:
Its ok, order of MANY_STUDENTS doesn't matter.

Reply:
Sorting rn is not that important here, will be tested thoroughly in sort command tests